### PR TITLE
Fix CI for 1.44.0 bump

### PR DESCRIPTION
**Pull request by milesziemer**
A test depended on the ordering of validation events, which 1.44.0 changes.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by syall — APPROVED** [review]

File: src/test/java/software/amazon/smithy/lsp/SmithyInterfaceTest.java [hunk 1]
```diff
@@ -105,10 +105,16 @@ public void runValidators() throws Exception {
...
         assertFalse(validationEvents.isEmpty());
-        assertEquals(
-            "Proto index 1 is used muliple times in members name,age of shape (structure: `some.test#MyStruct`).",
-                validationEvents.get(0).getMessage()
-        );
+
+        String expectedMessage = "Proto index 1 is used muliple times in members name," +
+                "age of shape (structure: `some.test#MyStruct`).";
+        Optional<ValidationEvent> matchingEvent = validationEvents.stream()
+                .filter(ev ->ev.getMessage().equals(expectedMessage)).findFirst();
```
Comment (hpmellema) [inline src/test/java/software/amazon/smithy/lsp/SmithyInterfaceTest.java:112]: nit: I think you could a hamcrest matcher (or `orElseThrow`) to make this a little cleaner
```
assertThat(validationEvents, hasItem(hasProperty("message", equals(expectedMessage)))
```
or
```
.orElseThrow(() -> new AssertionError(...))
```

**Review by hpmellema — APPROVED** [review]

File: src/test/java/software/amazon/smithy/lsp/SmithyInterfaceTest.java [hunk 2]
```diff
@@ -105,10 +105,16 @@ public void runValidators() throws Exception {
         assertTrue(result.isRight());
         List<ValidationEvent> validationEvents = result.getRight().getValidationEvents();
         assertFalse(validationEvents.isEmpty());
-        assertEquals(
-            "Proto index 1 is used muliple times in members name,age of shape (structure: `some.test#MyStruct`).",
-                validationEvents.get(0).getMessage()
-        );
+
+        String expectedMessage = "Proto index 1 is used muliple times in members name," +
```
Comment (sugmanue) [inline src/test/java/software/amazon/smithy/lsp/SmithyInterfaceTest.java:109]: Nit, there is a typo on "muliple" and a missing white space after the comma.

File: src/test/java/software/amazon/smithy/lsp/SmithyInterfaceTest.java (hunk 1 above)
Comment (milesziemer) [reply src/test/java/software/amazon/smithy/lsp/SmithyInterfaceTest.java:112]: I would if we used hamcrest in the language server. Will be adding it in the future though.

File: src/test/java/software/amazon/smithy/lsp/SmithyInterfaceTest.java (hunk 2 above)
Comment (milesziemer) [reply src/test/java/software/amazon/smithy/lsp/SmithyInterfaceTest.java:109]: Yea, that's just the message the validator produces, which lives in a jar in test resources. I don't remember where exactly it came from

**Pull request merged**